.github/workflows/deploy-gh-pages.yml

```diff
@@ -0,0 +1,71 @@
+name: Deploy to GitHub pages
+
+on:
+  push:
+    branches: [ "master" ]
+  workflow_dispatch:
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "gh-pages"
+  cancel-in-progress: false
+
+permissions: {}
+
+jobs:
+  build:
+    name: Build site
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'silverstripe'
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Read .nvmrc
+        id: read-nvm
+        shell: bash
+        run: echo "version=$(cat .nvmrc)" >> "$GITHUB_OUTPUT"
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ steps.read-nvm.outputs.version }}
+
+      - name: Setup GH pages
+        id: pages
+        uses: actions/configure-pages@v5
+
+      - name: Build static files
+        shell: bash
+        run: |
+          cat << EOF > .env
+          VUE_APP_GRAPHQL_ENDPOINT=https://api.github.com/graphql
+          VUE_APP_GRAPHQL_TOKEN=${{ secrets.GRAPHQL_TOKEN_FOR_GH_PAGES }}
```

Comment:
I've moved the same token that Vercel was using into the secrets for this repository.
Note that this token has always been made public by being included in the dist *.js files, so there's no change in attack surface by including it here like this. The .env file isn't included in the artifacts and therefore isn't in the GitHub Pages result - but the token will will be in the dist *.js files as it always has been.